### ISSUE TITLE
[Also a BUG REPORT] Should hvd.load_model be called on all ranks when resuming from a checkpoint?

### ISSUE THREAD

**Issue by thecooltechguy** [horovod/horovod#1964]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) Tensorflow 2.1.0's tf.keras
2. Framework version: 2.1.0
3. Horovod version: 18-19.2 (all versions in this range don't work)
4. MPI version: 4.0+
5. CUDA version: see note below
6. NCCL version: see note below
7. Python version: 3.6
8. OS and version: ubuntu
9. GCC version:

Note: I'm currently using AWS Sagemaker's default tensorflow environment (https://github.com/aws/sagemaker-tensorflow-container/blob/tf-2/docker/2.1.0/py3/Dockerfile.cpu).

**Checklist:**
1. Did you search issues to find if somebody asked this question before? Yes
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)? Yes
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)? Yes
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)? Yes

**Your question:**
Are we supposed to call `hvd.load_model()` on all ranks when resuming from a checkpoint, or are we only supposed to call it on rank 0 and let the broadcastvariableshook callback share these weights with the other workers? Is approach 1 incorrect/invalid, in that it will produce different results than approach 2?

I'm currently training a ResNet-based model with some BatchNorm layers, and if we only try to load the model on the first rank (and build/compile the model on the other ranks), we get a stalled tensor issue (which has been brought up here by someone else: #1271). However, if we call hvd.load_model on all ranks when resuming, training resumes normally but it seems to get immediately stuck and starts diverging, so we were confused as to whether loading the checkpoint model on all ranks (with hvd.load_model) can somehow cause training to diverge? But at the same time, we're unable to only load it on rank 0 because of #1271 causing Batch Norm to hang in horovod. Can someone please provide some tips here?

Thanks!

**Comment by tgaddair** [comment]
Hey @thecooltechguy, either approach should work in practice.  We usually recommend loading only on rank 0 in cases where you may not have the model checkpoint available to every worker.  If you load the model on all ranks, I would still do the broadcast, in case there was any random initialization during the load process.  Is this what you're doing?

If you have simple example that will repro the issue, we can definitely take a look.
